prov/lnx/src/lnx_init.c

```diff
@@ -171,75 +174,87 @@ struct util_prov lnx_util_prov = {
 	.flags = 0
 };
 
-/*
- * For the fi_getinfo() -> fi_fabric() -> fi_domain() path, we need to
- * keep track of the fi_info in case we need them later on when linking in
- * the fi_fabric() function.
- *
- * This cache gets cleared after we use the ones we need, or when the
- * library exists, if LNX is never used.
- */
-struct dlist_entry lnx_fi_info_cache;
 /* this is a list of all possible links */
 struct dlist_entry lnx_links;
-struct dlist_entry lnx_links_meta;
 
 struct lnx_fi_cache_entry {
 	struct dlist_entry entry;
 	struct fi_info *fi;
 };
 
-struct lnx_fi_info_meta {
+struct lnx_link_info {
 	struct dlist_entry entry;
-	struct fi_info *lnx_rep;
-	struct fi_info *lnx_link;
+	struct dlist_entry link_providers;
+	struct fi_info *fi_link;
 };
 
-static int lnx_get_cache_meta(struct dlist_entry *head, int *size)
+static void lnx_trim(char *str)
 {
-	int num_prov = 0;
-	struct dlist_entry *e;
+	int i, start, end, len;
 
-	dlist_foreach(head, e)
-		num_prov++;
+	if (!str)
+		return;
 
-	*size = num_prov;
+	len = strlen(str);
 
-	return FI_SUCCESS;
-}
+	for (start = 0; start < len && isspace(str[start]); ++start);
+	for (end = len - 1; end >= 0 && isspace(str[end]); --end);
```

Comment:
I think style convention is to put the semicolon on a new line to be clear that it's supposed to be empty? I'm not positive though - @j-xiong

Reply:
that's right.